tests/application/cms/test_views.py

```diff
@@ -806,99 +665,108 @@ def test_dept_user_should_not_be_able_to_edit_upload_if_page_not_shared(
+        uploads__title="upload title",
     )
 
-    assert response.status_code == 403
-
-    response = test_app_client.post(
+    response = test_app_client.get(
         url_for(
-            "cms.delete_dimension",
-            topic_slug=stub_page_with_dimension.measure.subtopic.topic.slug,
-            subtopic_slug=stub_page_with_dimension.measure.subtopic.slug,
-            measure_slug=stub_page_with_dimension.slug,
-            version=stub_page_with_dimension.version,
-            dimension_guid=stub_page_with_dimension.dimensions[0].guid,
+            "cms.edit_upload",
+            topic_slug=measure_version.measure.subtopic.topic.slug,
+            subtopic_slug=measure_version.measure.subtopic.slug,
+            measure_slug=measure_version.measure.slug,
+            version=measure_version.version,
+            upload_guid="test-download",
         )
     )
 
-    assert response.status_code == 403
+    assert response.status_code == 200
+    page = BeautifulSoup(response.data.decode("utf-8"), "html.parser")
+    assert page.find("h1").string == "Edit source data"
 
 
-def test_dept_user_should_be_able_to_edit_shared_page(
-    db_session, test_app_client, stub_measure_page, mock_logged_in_dept_user
-):
-    stub_measure_page.title = "this will be updated"
-    stub_measure_page.measure.shared_with.append(mock_logged_in_dept_user)
-    db_session.session.add(stub_measure_page)
-    db_session.session.commit()
+def test_dept_user_should_be_able_to_edit_shared_page(test_app_client, logged_in_dept_user):
+    measure_version = MeasureVersionFactory(
+        status="DRAFT", measure__shared_with=[logged_in_dept_user], title="this will be updated"
+    )
 
-    data = {"title": "this is the update", "db_version_id": stub_measure_page.db_version_id + 1}
+    data = {"title": "this is the update", "db_version_id": measure_version.db_version_id + 1}
     response = test_app_client.post(
         url_for(
             "cms.edit_measure_version",
-            topic_slug=stub_measure_page.measure.subtopic.topic.slug,
-            subtopic_slug=stub_measure_page.measure.subtopic.slug,
-            measure_slug=stub_measure_page.slug,
-            version=stub_measure_page.version,
+            topic_slug=measure_version.measure.subtopic.topic.slug,
+            subtopic_slug=measure_version.measure.subtopic.slug,
+            measure_slug=measure_version.measure.slug,
+            version=measure_version.version,
         ),
         data=data,
         follow_redirects=True,
     )
 
     assert response.status_code == 200
-
     page = BeautifulSoup(response.data.decode("utf-8"), "html.parser")
     assert page.find("div", class_="alert-box").span.string == 'Updated page "this is the update"'
+    assert measure_version.title == "this is the update"
+
+
+def test_dept_user_should_not_be_able_to_delete_dimension_if_page_not_shared(test_app_client, logged_in_dept_user):
+    measure_version = MeasureVersionWithDimensionFactory()
```

Comment:
Maybe an explicit `measure__shared_with=[]` here would be more clear? It's possibly OTT though - up to you.